src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -56,6 +57,16 @@ public ControllerModel BuildControllerModel([NotNull] TypeInfo typeInfo)
                 }
             }
 
+            foreach (var propertyInfo in controllerType.GetProperties(BindingFlags.Public | BindingFlags.Instance))
```

Comment:
did you want to isolate the way properties are fetched? Its similar to how we fetch action parameters. If user needs to override he already has `CreatePropertyModel`.